### ISSUE TITLE
`reportUnknownMemberType` false positive when accessing attribute inside lambda if the attribute is declared in supertype and modified in the subtype using an in-place operator

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1660]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnannotatedClassAttribute=false&code=MYGwhgzhAEBiD28BcAoa7oBMCmAzaA%2BgQJYB2xALkQBQTYi4CU0AtAHzQBy8p2qGA6HQYA6AG5gQAV2zQAvNACM0AMTQwmTGQDm66BQCeAB1lhSpeBTAViPaAAtsAJ1m5iAD2wxKKFKEgwAEJgTtQI8Iz8GDj4EPAAttgU9jq09EysHNy8UYLQ4PEARphgSELp4pIyqtDOTvBOZS5GDRQAqqQA1hYA7qQAsthFzgAqxthoeeWiEtKyANQKymou8fBiOvopMEYhFOogcdBunt4UQA)

```python
class Foo:
    def __init__(self) -> None:
        self.value = 1 # adding a type annotation here fixes it

class Bar(Foo):
    def something(self) -> None:
        lambda: self.value # error: reportUnknownMemberType
        self.value += 1 # removing this part also fixes it
```
